purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1823,7 +1830,8 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
                     observerMode,
                     platformInfo,
                     proxyURL,
-                    store
+                    store,
+                    autoSync = !configuration.dangerousSettings.contains(DangerousSetting.AUTO_SYNC_OFF)
```

Comment:
i don't want to block this PR if we are in a rush...but i wonder if in the future we might prefer dangerous settings to be a dictionary instead, so we can set values rather than just on/off...

Reply:
You got a great point there. It's an easy change. I don't know how to type the values of the map tho 🧐

Reply:
hmmm yeah we could do `Any` and then cast, and error if it's the wrong type? or just make dangerous settings a class with properties for each...